roachtest: ruby-pg failed

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.ruby-pg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/21097930?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/21097930?buildTab=artifacts#/ruby-pg) on master @ [1cc0b55ed8c8c9d42fbdad2d2ebcc1df973356c6](https://github.com/cockroachdb/cockroach/commits/1cc0b55ed8c8c9d42fbdad2d2ebcc1df973356c6):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.2.0-alpha.00000000-dev-1cc0b55ed8c8c9d42fbdad2d2ebcc1df973356c6
Tests run against ruby-pg v1.4.6
898 Total Tests Run
725 tests passed
173 tests failed
0 tests skipped
2 tests ignored
0 tests passed unexpectedly
2 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: Basic type mapping PG::BasicTypeMapForResults should warn about undefined types in format 0 -  (unexpected)
--- FAIL: PG::Tuple should override #inspect -  (unexpected)
For a full summary look at the ruby-pg artifacts 
An updated blocklist (rubyPGBlocklist) is available in the artifacts' ruby-pg log
test artifacts and logs in: /artifacts/ruby-pg/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-21097930-1770880338-170-n1cpu4-0001 | 104.196.6.171 | 10.142.1.32 |

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=epoch</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=true</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-21097930/ruby-pg/1770912586785/1770913185637)

</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*ruby-pg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60219

--- a/pkg/cmd/roachtest/tests/ruby_pg.go
+++ b/pkg/cmd/roachtest/tests/ruby_pg.go
@@ -266,11 +266,14 @@ func registerRubyPG(r registry.Registry) {
 	}
 
 	r.Add(registry.TestSpec{
-		Name:             "ruby-pg",
-		Timeout:          1 * time.Hour,
-		Owner:            registry.OwnerSQLFoundations,
-		Cluster:          r.MakeClusterSpec(1),
-		Leases:           registry.MetamorphicLeases,
+		Name:    "ruby-pg",
+		Timeout: 1 * time.Hour,
+		Owner:   registry.OwnerSQLFoundations,
+		Cluster: r.MakeClusterSpec(1),
+		Leases:  registry.MetamorphicLeases,
+		// Never run with runtime assertions as the overhead can delay
+		// sqlliveness heartbeats enough to cause session expiration errors.
+		CockroachBinary:  registry.StandardCockroach,
 		NativeLibs:       registry.LibGEOS,
 		CompatibleClouds: registry.OnlyGCE,
 		Suites:           registry.Suites(registry.Nightly, registry.Driver),

